common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java

```diff
@@ -22,7 +22,325 @@
+        mLoginHint = loginHint;
+    }
+
+    public String getUid() {
+        return mUid;
+    }
+
+    public void setUid(final String uid) {
+        mUid = uid;
+    }
+
+    public String getUtid() {
+        return mUtid;
+    }
+
+    public void setUtid(final String utid) {
+        mUtid = utid;
+    }
+
+    public String getDisplayableId() {
+        return mDisplayableId;
+    }
+
+    public void setDisplayableId(final String displayableId) {
+        mDisplayableId = displayableId;
+    }
+
+    public MicrosoftStsPromptBehavior getPromptBehavior() {
+        return mPromptBehavior;
+    }
+
+    public void setPromptBehavior(final MicrosoftStsPromptBehavior promptBehavior) {
+        mPromptBehavior = promptBehavior;
+    }
+
+    public PKCEChallenge getPKCEChallenge() {
+        return mPKCEChallenge;
+    }
+
+    public void setPKCEChallenge(final PKCEChallenge pKCEChallenge) {
+        mPKCEChallenge = pKCEChallenge;
+    }
+
+    public String getExtraQueryParam() {
+        return mExtraQueryParam;
+    }
+
+    public void setExtraQueryParam(final String extraQueryParam) {
+        mExtraQueryParam = extraQueryParam;
+    }
+
+    public String getSliceParameters() {
+        return mSliceParameters;
+    }
+
+    public void setmSliceParameters(final String sliceParameters) {
+        mSliceParameters = sliceParameters;
+    }
+
+    public URL getAuthority() {
+        return mAuthority;
+    }
+
+    public void setAuthority(final URL authority) {
+        mAuthority = authority;
+    }
+
+    /**
+     * @return String of MSAL version name.
+     */
+    public String getLibraryVersion() {
+        return mLibraryVersion;
+    }
+
+    /**
+     * The calling MSAL library should pass in its version name.
+     *
+     * @param libraryVersion String of the MSAL library version name.
+     */
+    public void setLibraryVersion(final String libraryVersion) {
+        mLibraryVersion = libraryVersion;
+    }
+
+    public String getAuthorizationStartUrl() throws UnsupportedEncodingException, ClientException{
+        if(AuthorizationConfiguration.getInstance().isBrokerRequest()) {
+            return getBrokerAuthorizationStartUrl();
+        } else {
+            return getLocalAuthorizationStartUrl();
+        }
+    }
+
+    private String getBrokerAuthorizationStartUrl() {
+        throw new UnsupportedOperationException("Not implemented.");
+
+    }
+
+    private String getLocalAuthorizationStartUrl() throws UnsupportedEncodingException, ClientException {
+        String authorizationUrl = StringExtensions.appendQueryParameterToUrl(
+                mAuthorizationEndpoint,
+                createAuthorizationRequestParameters());
+        Logger.infoPII(TAG, mCorrelationId.toString(), "Request uri to authorize endpoint is: " + authorizationUrl);
+        return authorizationUrl;
+    }
+
+    private Map<String, String> createAuthorizationRequestParameters() throws UnsupportedEncodingException, ClientException {
```

Comment:
I feel like I've read this already... is this copy/paste from the other class?

Reply:
Shrink this method. Break out into little separate functions. Share in a superclass where possible.